core/blockly.js

```diff
@@ -399,6 +399,33 @@ Blockly.prompt = function(message, defaultValue, callback) {
   callback(window.prompt(message, defaultValue));
 };
 
+/**
+ * Helper function for defining a block from json.  The resulting function has
+ * the correct value of jsonDef at the point in code where jsonInit is called.
+ * @param {Object} jsonDef The JSON definition of a block.
+ * @return {function} A function that calls jsonInit with the correct value
+ *     of jsonDef.
+ */
+Blockly.jsonInitFactory_ = function(jsonDef) {
+  return function() {
+    this.jsonInit(jsonDef);
+  };
+};
+
+/**
+ * Define blocks from an array of JSON block definitions, as might be generated
+ * by the Blockly Developer Tools.
+ * @param {!Array<Object>} jsonArray An array of JSON block definitions.
```

Comment:
The style guide recently make dots optional in array descriptors, but Blockly is currently consistent in using dots.  I have no objection if we remove all dots, but let's be consistent.  Also, !Object.